console/src/tasks.rs

```diff
@@ -115,13 +116,21 @@ impl State {
                 proto::tasks::task::Kind::Spawn => "T",
                 proto::tasks::task::Kind::Blocking => "B",
             };
-
+            let fields = task
+                .fields
+                .iter()
+                .fold(String::new(), |mut res, f| {
+                    write!(&mut res, "{} ", f).unwrap();
+                    res
+                })
+                .trim_end()
+                .into();
             let id = task.id?.id;
             let stats = stats_update.remove(&id)?.into();
             let task = Task {
                 id,
                 id_hex: format!("{:x}", id),
-                fields: task.string_fields,
+                fields,
```

Comment:
:+1: for now, since the purpose of this change was just to add structured fields to the wire format. 

a good follow-up would be to change the console TUI to store task fields as structured values as well (maybe a hashmap of names -> values for starters?). that would enable some additional things we could do, such as sorting the task list by the values of various fields, filtering which fields are displayed, and nicer formatting (e.g. bolding field names or displaying them in a different color).